src/Microsoft.Health.Fhir.Core/Features/Security/Authorization/RoleBasedFhirAuthorizationService.cs

```diff
@@ -53,7 +54,14 @@ public ValueTask<DataActions> CheckAccess(DataActions dataActions, CancellationT
                 }
             }
 
-            return new ValueTask<DataActions>(dataActions & permittedDataActions);
+            if (applySMARTAcess)
+            {
+                return new ValueTask<DataActions>(dataActions & permittedDataActions & SMARTScopeFhirAuthorizationService.CheckSMARTScopeAccess(_requestContextAccessor, dataActions));
+            }
+            else
```

Comment:
nit: else block is not required we could simply have
 return new ValueTask<DataActions>(dataActions & permittedDataActions);
after if condition

Reply:
fixed :)